syntax = "proto3";
package aumos.models.v1;

import "google/protobuf/timestamp.proto";
import "google/protobuf/struct.proto";

message Model {
  string id = 1;
  string tenant_id = 2;
  string name = 3;
  string version = 4;
  string framework = 5;
  ModelStatus status = 6;
  string artifact_uri = 7;
  google.protobuf.Struct metrics = 8;
  google.protobuf.Struct parameters = 9;
  repeated string tags = 10;
  string created_by = 11;
  google.protobuf.Timestamp created_at = 12;
  google.protobuf.Timestamp updated_at = 13;
}

enum ModelStatus {
  MODEL_STATUS_UNSPECIFIED = 0;
  DRAFT = 1;
  TRAINING = 2;
  VALIDATED = 3;
  DEPLOYED = 4;
  ARCHIVED = 5;
}
